Dockerfile

```diff
@@ -6,10 +6,17 @@ COPY . .
 # just use that.  For now we work around this by copying a tarball.
 RUN make install DESTDIR=./instroot && tar -C instroot -cf instroot.tar .
 
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.19-openshift-4.13 AS rhel9-builder
+ARG TAGS=""
+WORKDIR /go/src/github.com/openshift/machine-config-operator
+COPY . .
+RUN make install DESTDIR=./instroot
+
 FROM registry.ci.openshift.org/ocp/4.13:base
 ARG TAGS=""
 COPY --from=builder /go/src/github.com/openshift/machine-config-operator/instroot.tar /tmp/instroot.tar
 RUN cd / && tar xf /tmp/instroot.tar && rm -f /tmp/instroot.tar
+COPY --from=rhel9-builder /go/src/github.com/openshift/machine-config-operator/instroot/usr/bin/machine-config-daemon /usr/bin/machine-config-daemon
```

Comment:
Wait but we also use the binary as a container too; so now we're building in a rhel9 buildroot but running on top of rhel8, which doesn't make sense.

It probably does make sense to match the host but only for the binary copied to the host, so we'd need to move this to e.g. `/usr/bin/machine-config-daemon.rhel9` and then change the `copySelfToHost` logic to use that instead of `/proc/self/exe`.

Reply:
> That said if selfCopyToHost is removed, it's probably good to also remove the binary from on-disk there so as to not keep an outdated binary on the system

It lives in `/run`, so it will be automatically removed on the next boot